apps/search/js/providers/suggestions.js

```diff
@@ -59,6 +69,8 @@
     },
 
     search: function(input, preventRemote) {
+
+      this.currentSearch = input;
```

Comment:
Or if this is only a problem with clearing, I guess we could implement that method, and null the currentSearch on clear.